core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -111,9 +111,15 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) extends NonEmptyCollec
   def ++[AA >: A](l: List[AA]): NonEmptyList[AA] =
     concat(l)
 
+  def ++[AA >: A](other: Option[AA]): NonEmptyList[AA] =
+    concat(other)
+
   def concat[AA >: A](other: List[AA]): NonEmptyList[AA] =
     NonEmptyList(head, tail ::: other)
 
+  def concat[AA >: A](other: Option[AA]): NonEmptyList[AA] =
+    concat(other.toList)
```

Comment:
Thanks for the PR! Indeed we could maybe do something here like:
```suggestion
  def concat[F[_]: Foldable, AA >: A](other: F[AA]): NonEmptyList[AA] =
    concat(other.toList)
```

But FWIW I'm not sure how much this method really adds, seems easy enough to just call `.toList`?

Reply:
yeah, I'm -1 on the `Option[AA]` and `F[AA]` variants. I'd rather just do `.toList` and use the `Foldable` syntax there.